src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -102,7 +102,8 @@ public async Task InitializeAsync()
                 isFullyLoadedBool)
             {
                 _solutionOpened = true;
-                _threadHandling.Value.RunAndForget(() => CheckCompatibilityAsync(), unconfiguredProject: null);
+                // do not block package initialization on this
+                _ = CheckCompatibilityAsync();
```

Comment:
My understanding of this is admittedly lacking, but is there a "real fix" that can be done in the RunAndForget method, or the fault handling service it uses?

Reply:
RunAndForgot will work if you don't pass "CancelOnUnload". I would change RunAndForgot to remove CancelOnUnload if project is null.

Reply:
Yep that is the correct fix. One second

Reply:
You did this a different way by adding a new overload. The underlying bug is still there and it was deliberate that overload didn't exist; it forced us to pass a project if one was present. It results a much better error experience. Otherwise, I'm concerned that folks will opt into using this overload regardless of the layer they are in - leading to errors that don't indicate the project that was at fault.

Reply:
> The underlying bug is still there 

Which bug is still present?

> it was deliberate that overload didn't exist; it forced us to pass a project if one was present. It results a much better error experience.

I am ok with changing this back and special casing this singular case.

Reply:
The bug in the RunAndForgot method, if no project is passed, it still says to CancelOnUnload.

Reply:
alright, addressed here: https://github.com/dotnet/project-system/pull/4685